packages/client/src/v2-events/features/workqueues/EventOverview/components/EventHistory/EventHistory.tsx

```diff
@@ -185,6 +185,37 @@ export function EventHistory({ fullEvent }: { fullEvent: EventDocument }) {
   )
 
   const historyRows = visibleHistory
+    .reduce(
+      (actionList, currentAction) => {
+        const previousAction = actionList.at(-1)
+
+        // If correction request is approved by same user who requested it, we merge the actions
+        // and label it as 'CORRECT' action (which is an UI only action).
+        if (
+          previousAction &&
+          currentAction.type === ActionType.APPROVE_CORRECTION &&
+          previousAction.type === ActionType.REQUEST_CORRECTION &&
```

Comment:
Comparing only to the previous action feels a bit sketchy, as it might brake in some edge cases (for example low connectivity situation where some action, for example a read-action, might come between the request and approval) 🤔 

Also determining this with the `createdBy` feels a little bit hackish, but I understand why you ended up doing that 😊 

I'm thinking if we should add some kind of a boolean flag to the `APPROVE_CORRECTION` action & API request to mark it as a 'direct correction', such as `isImmediateCorrection` or `isDirectCorrection`, which could then be used to filter out the `REQUEST_CORRECTION` action?